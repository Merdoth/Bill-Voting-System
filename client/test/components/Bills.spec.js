import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Bills } from
  '../../js/components/Bills';

window.$ = require('jquery');

const props = {
  bills: [
    {
      _id: '5ab8e269f10a5e001481c955',
      updated_at: '2018-03-26T12:07:05.593Z',
      created_at: '2018-03-26T12:07:05.593Z',
      title: 'JAMB OFFICIALS ',
      description: 'The stole money',
      billProgress: 'Not enacted',
      downVoteCount: 0,
      upVoteCount: 0
    }
  ],
  user: {
    token: {}
  },
  getAllBills: jest.fn(),
  allBills: [],
  currentPage: {}
};

const setup = () => shallow(<Bills {...props} />);

describe('Bills Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('handlePageClick()', () => {
    describe('handlePageClick()', () => {
      it('should get the next all users', () => {
        const wrapper = setup();
        wrapper.instance().handlePageClick({
          selected: 3
        });
        expect(props.getAllBills).toHaveBeenCalledWith(4);
      });
    });
  });
});

