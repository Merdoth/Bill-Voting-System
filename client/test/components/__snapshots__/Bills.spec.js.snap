// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bills Component snapshot it should render the right amount of elements 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      >
        <Link
          replace={false}
          to="/search"
        >
          <i
            className="material-icons medium search-icon"
          >
            search
          </i>
        </Link>
      </span>
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="flexer flexer-row"
    >
      <div
        className="flexed bill-list"
      >
        <div
          className="bill-header"
        >
          <h3>
            Bills
          </h3>
        </div>
        <div
          className="bill-body"
        >
          <h3>
            Sadly nothing to show :)
          </h3>
          <PaginationBoxView
            activeClassName="active"
            breakLabel="..."
            containerClassName="pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={1}
            nextClassName="next"
            nextLabel="next"
            onPageChange={[Function]}
            pageCount={0}
            pageRangeDisplayed={5}
            previousClassName="previous"
            previousLabel="previous"
            subContainerClassName="pages pagination"
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
